trace/trace.go

```diff
@@ -65,12 +79,19 @@ func (t *Trace) logWithStepThreshold(stepThreshold time.Duration) {
 	endTime := time.Now()
 
 	totalTime := endTime.Sub(t.startTime)
-	buffer.WriteString(fmt.Sprintf("Trace[%d]: %q (started: %v) (total time: %v):\n", tracenum, t.name, t.startTime, totalTime))
+	buffer.WriteString(fmt.Sprintf("Trace[%d]: %q ", tracenum, t.name))
```

Comment:
I wonder if we can use `strings.Builder` to replace `fmt.Sprintf` https://stackoverflow.com/a/47798475

Reply:
I'm a bit concerned with the loss of readability for thins like "Trace[%d]:" if we used a string builder. But I have made heavier use of buffer.WriteString where appropriate.